src/docs/arch-hs.dox

```diff
@@ -1,8 +1,272 @@
+updated once the prepared update is resolved. A tombstone becomes globally visible if the stop
+timestamp is less than the oldest timestamp and the stop transaction id is visible to all concurrent
+transactions in the system. A checkpoint can delete history store pages that only contain globally
+visible tombstones. The garbage collection process is discussed in @ref arch-checkpoint.
+
+@section arch_hs_initialize History store initialization
+WiredTiger checks for an existing history store table at startup and creates a new table if it is
+not able to find one. The history store table is usually created when WiredTiger creates a
+new database, with one exception where WiredTiger is opening an existing database created by an
+older version of WiredTiger that doesn't support the history store. WiredTiger uses
+\c WiredTigerHS.wt as the filename for the history store table.
+
+@section arch_hs_cursor History store cursor interface
+WiredTiger uses a cursor interface for history store table to make it easier for different modules
+to perform data operations on the history store table. A new cursor can be opened by calling
+\c __wt_curhs_open(). History store cursor implementation supports most of the standard \c WT_CURSOR
+methods except for \c WT_CURSOR::search() method. Recall that the key contains a
+counter and timestamp value and it is expected that the caller can not possibly know the exact
+values for both fields beforehand. Therefore, \c WT_CURSOR::search_near() is used for all search
+operations. Two helper functions, \c __wt_curhs_search_near_before() and
+\c __wt_curhs_search_near_after(),
+have been provided to facilitate searching for the required record in the history store table.
+
+@section arch_hs_cursor_visibility History store cursor interface and visibility rules
+When using the history store cursor interface, the user can configure the type of visibility checks
+that are performed on the records. The behavior is controlled by a set of cursor flags:
+
+- By default, snapshot based visibility checks are performed on all records returned to the API
+user.
+
+- When the flag \c WT_CURSTD_HS_READ_ALL is set on the cursor, no visibility checks are performed
+on the records returned to the API user. This means that cursor interface can even return a record
+with a globally visible tombstone. When this flag is set, it suppresses the effect of other
+visibility flags.
+
+- When the flag \c WT_CURSTD_HS_READ_COMMITTED is set, the cursor interface can return any record
+except for globally deleted records. This flag has no effect when the flag
+\c WT_CURSTD_HS_READ_ALL is set.
+
+History store cursor users must use one of the flags if the caller thread is running in a lower
+isolation level and doesn't hold a valid snapshot.
+
+@section arch_hs_reconciliation History store and reconciliation
+When a dirty page is reconciled on a user file b-tree, the update chain is examined and the latest
```

Comment:
Sometimes you write `b-tree` and sometimes `btree`, we should agree on one spelling and have it consistent.

Reply:
I have changed spellings to `btree` to make it consistent across the document.